Enhance developer agent documentation with new features (#8943)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-ai/developer-agent.md

# Developer agent <Lifecycle status="beta,managed,managed_plus"/>

<IntroText>
The <Constant name="dev_agent" /> is the next evolution of <Constant name="copilot" /> and provides agentic capabilities to streamline the developer experience in the <Constant name="studio_ide" />. Build, refactor, test, document, and explore your dbt project from natural language &mdash; grounded in your project's lineage, metadata, governance, and <Constant name="semantic_layer" /> &mdash; while keeping every change auditable.
</IntroText>
  
The <Constant name="dev_agent" /> is designed to support the full dbt development lifecycle &mdash; from initial model creation to testing, documentation, and semantic layer definition. Some examples of what you can do:

> - Build or modify models: Create or update dbt models from natural language descriptions of the transformation or logic you need.

Reviewer: These headings feel a little duplicative.
```suggestion
- Build or modify models from natural language descriptions of the transformation or logic you need.
```

> - Generate tests, docs, and semantic models: Scaffold YAML definitions from existing models, define and validate semantic metrics, and save time on manual setup.

Reviewer: ```suggestion
- Automatically generate and validate YAML definitions for tests, documentation, and semantic models directly from your existing dbt models, eliminating manual setup.
```

> - Refactor and maintain your project: Rename columns, change materializations, adjust logic, or add advanced data quality tests &mdash; the agent keeps associated YAML files in sync.

Reviewer: ```suggestion
- Make targeted changes to your project such as renaming columns, changing materializations, adjusting logic, or adding advanced data quality tests — while the agent keeps associated YAML files in sync.
```

> - Explore and query your project: Ask questions about your DAG, query <Constant name="catalog"/>, and surface lineage and metadata to inform your development.

Reviewer: ```suggestion
- Explore your project with the agent by asking questions about your DAG, querying the <Constant name="catalog"/>, and surfacing lineage and metadata to guide your development.
```

The agent comes with the following out of the box — no configuration needed! 🎉

- [dbt Agent Skills](https://github.com/dbt-labs/dbt-agent-skills): A curated collection of dbt-recommended guidance and instructions, managed by dbt Labs and always up to date.

> - [dbt MCP server Product docs toolset](/docs/dbt-ai/mcp-available-tools#product-docs): Enables the agent to search and fetch content directly from dbt's official documentation at docs.getdbt.com.

Author: ```suggestion
- [dbt MCP server Product docs toolset](/docs/dbt-ai/mcp-available-tools#product-docs): Enables the agent to search and fetch content directly from dbt's official documentation at [docs.getdbt.com](https://docs.getdbt.com/).
```
